packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -182,11 +182,56 @@ function EditableBlockBindingsPanelItems( {
 export const BlockBindingsPanel = ( { name: blockName, metadata } ) => {
 	const registry = useRegistry();
 	const blockContext = useContext( BlockContext );
-	const { bindings } = metadata || {};
 	const { removeAllBlockBindings } = useBlockBindingsUtils();
 	const bindableAttributes = getBindableAttributes( blockName );
 	const dropdownMenuProps = useToolsPanelDropdownMenuProps();
 
+	// While this hook doesn't directly call any selectors, `useSelect` is
+	// used purposely here to ensure `getFieldsList` is updated whenever
+	// there are attribute updates.
+	// `source.getFieldsList` may also call a selector via `registry.select`.
+	const { fieldsList } = useSelect( () => {
```

Comment:
We should combine the other `useSelect` into this one: [link](https://github.com/WordPress/gutenberg/blob/bc6fae8c8d41763b2786fce8f228d96eb89b014c/packages/block-editor/src/hooks/block-bindings.js#L245-L250). I forgot to do so when I added it.